xla/service/collective_pipeliner.cc

```diff
@@ -2347,6 +2348,11 @@ StatusOr<bool> CollectivePipeliner::Run(
           << " and transformed instructions: " << transformed_instructions
           << " for pipelining direction: "
           << GetPipelineDirectionString(config_.pipelining_direction);
+  // Run necessary cleanup to make sure unused code doesn't trigger HloVerifier.
+  if (changed) {
+    TF_RETURN_IF_ERROR(HloDCE().Run(module, execution_threads).status());
```

Comment:
Can we run the pass using the pass pipeline? That way if we enable HLO dumping, we get the dumps.

Reply:
I think sometimes the hlo verifier will be run after every pass, if the erroneous case such as the one in the unit test shows up, it will error out right after this pass without running dce so we need to run it here as a sub-pass. Is there a way to run a pipeline within a pass or we can just print the module with a vlog here.

Reply:
Got it, then its fine as is I think. We will still get the dump post the pipeliner pass.